### ISSUE TITLE
Accept multiple elements for geographic reference

### ISSUE THREAD

**Issue by bondjimbond** [Islandora-Labs/islandora_simple_map#2]
Include in the config form additional (if not then..) options for elements that could contain a geographic reference.

Example: some objects may contain a geographic subject, but no cartographic coordinates. Others may have coordinates but no geographic subject. Module should ideally look first at one element (specified in configuration), then at another, and optionally another, etc.

**Comment by mjordan** [comment]
Yes, this is possible. I'm thinking that the admin would list several XPaths in preferred order, and the first one to match would be used.

**Comment by mjordan** [comment]
@bondjimbond can you test the issue-2 branch, which now supports multiple source elements? Enter the following into the "XPath expressions to MODS elements containing map data" field in the module's admin settings:

```
//mods:subject/mods:cartographics/mods:coordinates
//mods:subject/mods:geographic
```

Then add data to both MODS elements in your test object(s). Test rendering the map, switch the order of the XPath expressions, remove one of them, etc. to see if the results are what you expect. Also take a look at the README to see if everything is explained clearly.

**Comment by bondjimbond** [comment]
Seems to work well. 

In the Basic Image Form, there's another MODS section that might be useful to include as a default XPath, though I don't know exactly how it could be written as there are so many options... 

//mods:subject/mods:hierarchicalGeographic - and then a load of other elements below, from very broad (continent) down to very nary (city section). For this implementation, would there have to be multiple lines, descending in order of priority? 

e.g. first line //mods:subject/mods:hierarchicalGeographic/mods:citySection, next line //mods:subject/mods:hierarchicalGeographic/mods:city, then //mods:subject/mods:hierarchicalGeographic/mods:county, broader at every line?

**Comment by mjordan** [comment]
Having separate xpaths would be one way to do it, but it might also be possible to create one large xpath expression that selected the first element that had content. But, the separate xpaths as you describe would allow the admin to determine what the priority is, to suit the site's metadata. 

Thanks for testing. I'll go ahead and merge this into 7.x. If you want to open a separate issue about //mods:subject/mods:hierarchicalGeographic that would be fine.

**Comment by mjordan** [comment]
Closed with 0ac6f8200ffd9e153716f9726b792860e45785b9.
